static/js/chart/draw.ts

```diff
@@ -431,42 +436,77 @@ function drawLineChart(
 
 
 /**
- * Draw a group of line charts with different goeId.
+ * Return all styles information for given dataGroupsDict.
+ * {
+ *     colors: [],
+ *     dashes: [],
+ *     statVars: [],
+ *     geoIds: [],
+ *     maxV: ,
+ * }
  *
- * @param id
- * @param width
- * @param height
  * @param dataGroupsDict
- * @param unit
  */
-function drawGroupLineChart(
-    id: string,
-    width: number,
-    height: number,
-    dataGroupsDict: {[geoId: string]: DataGroup[];},
-    unit?: string
+function generateStyles(
+    dataGroupsDict: {[geoId: string]: DataGroup[] },
 ) {
-  console.log(dataGroupsDict);
+  let styles = {};
+  styles["colors"] = [];
+  styles["dashes"] = [];
+  styles["statVars"] = [];
+  styles["geoIds"] = [];
+
+  let dataGroups : DataGroup[];
+  dataGroups = Object.values(dataGroupsDict)[0];
+  let legendText = dataGroups.map((dataGroup) => dataGroup.label ? dataGroup.label: '');
+  let colorFn = getColorFn(legendText);
 
-  let dataGroups;
+  for (let i = 0; i < dataGroups.length; i++) {
+    styles["colors"].push(colorFn(dataGroups[i].label));
+    styles["statVars"].push(dataGroups[i].label);
+  }
+
+  let dashIndex = 0;
+  let maxV = 0;
   for (let geoId in dataGroupsDict) {
     dataGroups = dataGroupsDict[geoId];
-    break;
+    maxV = Math.max(maxV, Math.max(...dataGroupsDict[geoId].map((dataGroup) => dataGroup.max())));
+    styles["geoIds"].push(geoId);
+    styles["dashes"].push(DASHES[dashIndex]);
+    dashIndex++;
```

Comment:
How should we handle too many groupings, more than we have predefined dashes for?

Consider adding a modulo here so we don't run out of dashes? Or is there a way to generate more dashes programatically?

Reply:
Yep! This is also what I am thinking to improve here.

Reply:
Done